### ISSUE TITLE
Low accuracy of the detection after Encodec transformation

### ISSUE THREAD

**Issue by ObsisMc** [facebookresearch/audioseal#27]
When testing Encodec transformation, I firstly encoded the watermarked audio and then decoded it and got the transformed audio. However, I couldn't detect a watermark on the transformed audio and the watermark probability was very small.

I am not sure whether my test is correct and could you provide some code to show how to test Encodec?

**Comment by pierrefdz** [comment]
Hi! Sorry for the delay in the answer!

By default we use this ckpt: "//pretrained/facebook/encodec_24khz" (with varying nq).
To use it, we use `CompressionSolver.model_from_checkpoint(ckpt)` from AudioCraft, see [EnCodec](https://github.com/facebookresearch/audiocraft/blob/main/docs/ENCODEC.md)

We noticed that the robustness is specific to the auto-encoders seen at train time, what was the model that you tested? (can you provide code to reproduce?)

**Comment by ObsisMc** [comment]
Thanks for your reply. The Encodec I used before was `facebook/encodec_32khz` as follows

```python
import soundfile as sf

device = "cuda" if torch.cuda.is_available() else "cpu"

# models
model = AudioSeal.load_generator("audioseal_wm_16bits").to(device)
detector = AudioSeal.load_detector(("audioseal_detector_16bits")).to(device)
encodec_model = CompressionModel.get_pretrained('facebook/encodec_32khz').to(device)

# load audio
audio_path = "audio_0.wav"
audio, sr = sf.read(audio_path)  # sr: 32000
audio = torch.tensor(audio).float()
audio = audio.to(device)
audios = audio.unsqueeze(0).unsqueeze(0)

# encode watermark
watermark = model.get_watermark(audios, sample_rate=sr)
watermarked_audio = audios + watermark

# Encodec attack
encode_audio = encodec_model.encode(watermarked_audio)
audio_codes, scaling = encode_audio[0], encode_audio[1]
watermarked_audio = encodec_model.decode(audio_codes, scaling)

# detect
result, message = detector.detect_watermark(watermarked_audio, sample_rate=sr, message_threshold=0.5)
print(f"\nWM probability: {result}")

```
where the output is almost 0 but I think the probability should be over 0.5.

Then I also tested `encodec_model = CompressionModel.get_pretrained('facebook/encodec_24khz').to(device)` as you mentioned and the probability indeed improved though it couldn't exceed 0.5 either.

So I need to find a good hyperparameter to get a better result?

**Comment by pierrefdz** [comment]
Hi @ObsisMc,
Have you followed up on this? Have you tried with encodec_24khz instead?

**Comment by ObsisMc** [comment]
Yes I tried it as I mentioned at the end of my last comment.

> Then I also tested encodec_model = CompressionModel.get_pretrained('facebook/encodec_24khz').to(device) as you mentioned and the probability indeed improved though it couldn't exceed 0.5 either.

I think maybe it is about the hyperparameters.
